reactjs-films-homework/src/components/MovieCard/MovieCard.tsx

```diff
@@ -24,42 +26,53 @@ export interface IGenre {
   }[];
 }
 
-const MovieCard: React.FC<ICard> = ({ props }) => {
-  const { id, title, vote_average, poster_path, genre_ids } = props;
+const MovieCard: React.FC<ICard> = ({
+  props: { id, title, vote_average, poster_path, genre_ids },
+}) => {
+  const imgWidth = '52px';
   const movieTitle = 'Movie Title';
   const urlImg = `https://image.tmdb.org/t/p/w500/${poster_path}`;
-  const [genre, setGenre] = useState<IGenre>({});
-  const [genreName, setGenreName] = useState<Array<string>>([]);
-
-  mockGenres.then((res) => {
-    return setGenre(res);
-  });
+  const [genreName, setGenreName] = useState<string[]>([]);
+  const dispatch = useAppDispatch();
+  const appFetchMovieGenre = useAppSelector((state) => state.mainReducer.genre);
 
-  useEffect(() => {
-    if (genre.genres) {
-      genre.genres.map((genre): void => {
-        genre_ids!.map((el): void => {
-          if (genre.id === el) {
-            setGenreName((prev) => {
-              prev.push(genre.name);
-              const set = new Set(prev);
-              const arr = Array.from(set);
-              prev = arr;
-              return prev;
-            });
-          }
-        });
+  useMemo(() => {
+    const arrToSort: string[] = [];
+    if (appFetchMovieGenre!.genres?.length) {
+      appFetchMovieGenre!.genres.map((genre): void => {
+        genre_ids?.length &&
+          genre_ids!.map((el): void => {
+            if (genre!.id === el) {
+              setGenreName((prev) => {
+                arrToSort.push(genre.name);
+                const set = new Set(arrToSort);
+                const arr = Array.from(set);
+                prev = arr;
+                return prev;
+              });
+            }
+          });
       });
     }
-  }, [genre, genre_ids]);
+  }, [genre_ids, appFetchMovieGenre]);
```

Comment:
Too complex logic. Do you know KISS principle? :)
`
const genreName = useMemo(() => {
    return genre_ids?.reduce((acc, curr) => {
      if (appFetchMovieGenre?.genres?.[curr]) {
        acc.push(appFetchMovieGenre?.genres?.[curr].name);
      }
      return acc;
    }, [] as string[]).sort();
  }, [genre_ids, appFetchMovieGenre]);
  
`
This way it will be better, but usually such logic should move to selectors level. It will be our next step.